db fails to build on amd64

The package db with version 5.3.28 and epoch 2 fails to build from source on amd64 with the following error:

```
2025-06-02 07:35:34.467801632 +0000 UTC src/log/log_archive.c: In function '__absname':
2025-06-02 07:35:34.467813789 +0000 UTC src/log/log_archive.c:565:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.467822948 +0000 UTC   565 | __absname(env, pref, name, newnamep)
2025-06-02 07:35:34.467827819 +0000 UTC       | ^~~~~~~~~
2025-06-02 07:35:34.467989431 +0000 UTC src/log/log_archive.c: In function '__usermem':
2025-06-02 07:35:34.467999677 +0000 UTC src/log/log_archive.c:600:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.46800911 +0000 UTC   600 | __usermem(env, listp)
2025-06-02 07:35:34.468015412 +0000 UTC       | ^~~~~~~~~
2025-06-02 07:35:34.468144156 +0000 UTC src/log/log_archive.c: In function '__cmpfunc':
2025-06-02 07:35:34.468153021 +0000 UTC src/log/log_archive.c:639:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.468157797 +0000 UTC   639 | __cmpfunc(p1, p2)
2025-06-02 07:35:34.468161487 +0000 UTC       | ^~~~~~~~~
2025-06-02 07:35:34.506061039 +0000 UTC src/log/log.c: In function '__log_open':
2025-06-02 07:35:34.506112184 +0000 UTC src/log/log.c:29:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.506118414 +0000 UTC    29 | __log_open(env)
2025-06-02 07:35:34.506122489 +0000 UTC       | ^~~~~~~~~~
2025-06-02 07:35:34.507291728 +0000 UTC src/log/log.c: In function '__log_init':
2025-06-02 07:35:34.507301703 +0000 UTC src/log/log.c:178:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.507307039 +0000 UTC   178 | __log_init(env, dblp)
2025-06-02 07:35:34.507310655 +0000 UTC       | ^~~~~~~~~~
2025-06-02 07:35:34.508171613 +0000 UTC src/log/log.c: In function '__log_recover':
2025-06-02 07:35:34.508181008 +0000 UTC src/log/log.c:289:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.508185353 +0000 UTC   289 | __log_recover(dblp)
2025-06-02 07:35:34.508189091 +0000 UTC       | ^~~~~~~~~~~~~
2025-06-02 07:35:34.508874212 +0000 UTC src/log/log.c: In function '__log_find':
2025-06-02 07:35:34.508883459 +0000 UTC src/log/log.c:411:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.508887756 +0000 UTC   411 | __log_find(dblp, find_first, valp, statusp)
2025-06-02 07:35:34.508891287 +0000 UTC       | ^~~~~~~~~~
2025-06-02 07:35:34.50997082 +0000 UTC src/log/log.c: In function '__log_valid':
2025-06-02 07:35:34.510000797 +0000 UTC src/log/log.c:589:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.510009084 +0000 UTC   589 | __log_valid(dblp, number, set_persist, fhpp, flags, statusp, versionp)
2025-06-02 07:35:34.510012763 +0000 UTC       | ^~~~~~~~~~~
2025-06-02 07:35:34.511150309 +0000 UTC src/log/log.c: In function '__log_env_refresh':
2025-06-02 07:35:34.511164114 +0000 UTC src/log/log.c:870:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.511168363 +0000 UTC   870 | __log_env_refresh(env)
2025-06-02 07:35:34.511176278 +0000 UTC       | ^~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.512654483 +0000 UTC src/log/log.c: In function '__log_get_cached_ckp_lsn':
2025-06-02 07:35:34.512664862 +0000 UTC src/log/log.c:994:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.512669586 +0000 UTC   994 | __log_get_cached_ckp_lsn(env, ckp_lsnp)
2025-06-02 07:35:34.512673269 +0000 UTC       | ^~~~~~~~~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.512876001 +0000 UTC src/log/log.c: In function '__log_region_mutex_count':
2025-06-02 07:35:34.512884026 +0000 UTC src/log/log.c:1018:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.512887961 +0000 UTC  1018 | __log_region_mutex_count(env)
2025-06-02 07:35:34.512891651 +0000 UTC       | ^~~~~~~~~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.512913087 +0000 UTC src/log/log.c: In function '__log_region_mutex_max':
2025-06-02 07:35:34.512921439 +0000 UTC src/log/log.c:1036:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.51292659 +0000 UTC  1036 | __log_region_mutex_max(env)
2025-06-02 07:35:34.512931643 +0000 UTC       | ^~~~~~~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.513022808 +0000 UTC src/log/log.c: In function '__log_region_size':
2025-06-02 07:35:34.513033188 +0000 UTC src/log/log.c:1061:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.513039476 +0000 UTC  1061 | __log_region_size(env)
2025-06-02 07:35:34.513043961 +0000 UTC       | ^~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.513155216 +0000 UTC src/log/log.c: In function '__log_region_max':
2025-06-02 07:35:34.513165417 +0000 UTC src/log/log.c:1086:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.513172405 +0000 UTC  1086 | __log_region_max(env)
2025-06-02 07:35:34.513177511 +0000 UTC       | ^~~~~~~~~~~~~~~~
2025-06-02 07:35:34.513302844 +0000 UTC src/log/log.c: In function '__log_vtruncate':
2025-06-02 07:35:34.513313254 +0000 UTC src/log/log.c:1118:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.513321075 +0000 UTC  1118 | __log_vtruncate(env, lsn, ckplsn, trunclsn)
2025-06-02 07:35:34.513327671 +0000 UTC       | ^~~~~~~~~~~~~~~
2025-06-02 07:35:34.514069798 +0000 UTC src/log/log.c: In function '__log_is_outdated':
2025-06-02 07:35:34.514078963 +0000 UTC src/log/log.c:1216:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.514083323 +0000 UTC  1216 | __log_is_outdated(env, fnum, outdatedp)
2025-06-02 07:35:34.514087071 +0000 UTC       | ^~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.514574896 +0000 UTC src/log/log.c: In function '__log_zero':
2025-06-02 07:35:34.514583058 +0000 UTC src/log/log.c:1277:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.514587146 +0000 UTC  1277 | __log_zero(env, from_lsn)
2025-06-02 07:35:34.514590063 +0000 UTC       | ^~~~~~~~~~
2025-06-02 07:35:34.515760787 +0000 UTC src/log/log.c: In function '__log_inmem_lsnoff':
2025-06-02 07:35:34.515769965 +0000 UTC src/log/log.c:1379:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.515773758 +0000 UTC  1379 | __log_inmem_lsnoff(dblp, lsnp, offsetp)
2025-06-02 07:35:34.515777203 +0000 UTC       | ^~~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.516030003 +0000 UTC src/log/log.c: In function '__log_inmem_newfile':
2025-06-02 07:35:34.516037928 +0000 UTC src/log/log.c:1407:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.516041775 +0000 UTC  1407 | __log_inmem_newfile(dblp, file)
2025-06-02 07:35:34.516045358 +0000 UTC       | ^~~~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.517448979 +0000 UTC src/log/log.c: In function '__log_inmem_chkspace':
2025-06-02 07:35:34.517458082 +0000 UTC src/log/log.c:1481:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.517461964 +0000 UTC  1481 | __log_inmem_chkspace(dblp, len)
2025-06-02 07:35:34.517465171 +0000 UTC       | ^~~~~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.518599752 +0000 UTC src/log/log.c: In function '__log_inmem_copyout':
2025-06-02 07:35:34.518608429 +0000 UTC src/log/log.c:1567:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.518612413 +0000 UTC  1567 | __log_inmem_copyout(dblp, offset, buf, size)
2025-06-02 07:35:34.518615976 +0000 UTC       | ^~~~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.518759873 +0000 UTC src/log/log.c: In function '__log_inmem_copyin':
2025-06-02 07:35:34.518769909 +0000 UTC src/log/log.c:1592:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.518775834 +0000 UTC  1592 | __log_inmem_copyin(dblp, offset, buf, size)
2025-06-02 07:35:34.518780596 +0000 UTC       | ^~~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.51887417 +0000 UTC src/log/log.c: In function '__log_set_version':
2025-06-02 07:35:34.518884093 +0000 UTC src/log/log.c:1619:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.518889361 +0000 UTC  1619 | __log_set_version(env, newver)
2025-06-02 07:35:34.518893707 +0000 UTC       | ^~~~~~~~~~~~~~~~~
2025-06-02 07:35:34.518926839 +0000 UTC src/log/log.c: In function '__log_get_oldversion':
2025-06-02 07:35:34.518937191 +0000 UTC src/log/log.c:1647:1: warning: old-style function definition [-Wold-style-definition]
2025-06-02 07:35:34.518942951 +0000 UTC  1647 | __log_get_oldversion(env, ver)
2025-06-02 07:35:34.518947626 +0000 UTC       | ^~~~~~~~~~~~~~~~~~~~
2025-06-02 07:35:40.167156545 +0000 UTC make: Leaving directory '/home/build'
2025-06-02 07:35:44.582638958 +0000 UTC failed to build package: unable to run package db pipeline: unable to run pipeline: unable to run pipeline: exit status 2
2025-06-02 07:35:44.621658397 +0000 UTC failed to build: command melange failed: exit status 1
2025-06-02 07:35:44.818919756 +0000 UTC build failed: command melange failed: exit status 1
```

--- a/db.yaml
+++ b/db.yaml
@@ -1,7 +1,7 @@
 package:
   name: db
   version: 5.3.28
-  epoch: 2
+  epoch: 3
   description: The Berkeley DB embedded database system
   copyright:
     - license: LicenseRef-berkeley-db
@@ -15,6 +15,7 @@ environment:
       - build-base
       - busybox
       - ca-certificates-bundle
+      - gcc-14-default
   environment:
     CFLAGS: "-Wno-error=implicit-int -Wno-error=implicit-function-declaration"
 

